[CI] DockerTests test011SecurityEnabledStatus failing

**Build Scans:**
- [elasticsearch-periodic-packaging #11959 / rocky-10_packaging-tests-unix](https://gradle-enterprise.elastic.co/s/fjh3iitrvhgok)
- [elasticsearch-periodic-packaging #11959 / ubuntu-2404_packaging-tests-unix](https://gradle-enterprise.elastic.co/s/qdj7v3i55cjxk)
- [elasticsearch-periodic-packaging #11959 / rocky-9_packaging-tests-unix](https://gradle-enterprise.elastic.co/s/n37docfz5g42a)
- [elasticsearch-periodic-packaging #11959 / rhel-9_packaging-tests-unix](https://gradle-enterprise.elastic.co/s/whosirhboguj2)
- [elasticsearch-periodic-packaging #11959 / rhel-9_packaging-tests-unix](https://gradle-enterprise.elastic.co/s/whosirhboguj2)
- [elasticsearch-pull-request #151437 / rhel-8_destructivedistrotest.docker_packaging-tests-unix-sample](https://gradle-enterprise.elastic.co/s/unstbok7ssj7e)
- [elasticsearch-pull-request #151389 / ubuntu-2404_destructivedistrotest.docker_packaging-tests-unix-sample](https://gradle-enterprise.elastic.co/s/53jes7bdnggpw)
- [elasticsearch-pull-request #151389 / rhel-8_destructivedistrotest.docker_packaging-tests-unix-sample](https://gradle-enterprise.elastic.co/s/zfh6y6froqixo)
- [elasticsearch-pull-request #151389 / ubuntu-2404_destructivedistrotest.docker_packaging-tests-unix-sample](https://gradle-enterprise.elastic.co/s/3m5jysrbpywuy)
- [elasticsearch-pull-request #151389 / rhel-8_destructivedistrotest.docker_packaging-tests-unix-sample](https://gradle-enterprise.elastic.co/s/pma3k4lsuflws)

**Reproduction Line:**
```
null
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.packaging.test.DockerTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('test011SecurityEnabledStatus'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: <401>
     but: was <503>
```

**Issue Reasons:**
- [main] 1 consecutive failures in step rocky-10_packaging-tests-unix
- [main] 1 consecutive failures in step rhel-9_packaging-tests-unix
- [main] 1 consecutive failures in pipeline elasticsearch-periodic-packaging
- [main] 17 failures in test test011SecurityEnabledStatus (3.0% fail rate in 565 executions)
- [main] 7 failures in step ubuntu-2404_destructivedistrotest.docker_packaging-tests-unix-sample (2.9% fail rate in 239 executions)
- [main] 5 failures in step rhel-8_destructivedistrotest.docker_packaging-tests-unix-sample (2.1% fail rate in 238 executions)
- [main] 6 failures in pipeline elasticsearch-pull-request (2.5% fail rate in 237 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -380,9 +380,6 @@ tests:
 - class: org.elasticsearch.xpack.ccr.CcrSeqNoPruningIT
   method: testSeqNoPartiallyRetainedByCcrLease
   issue: https://github.com/elastic/elasticsearch/issues/150435
-- class: org.elasticsearch.packaging.test.DockerTests
-  method: test011SecurityEnabledStatus
-  issue: https://github.com/elastic/elasticsearch/issues/150438
 - class: org.elasticsearch.xpack.esql.action.FileSourceSecretDecryptionAcrossNodesIT
   method: testJoinedDataNodeDecryptsTheSecret
   issue: https://github.com/elastic/elasticsearch/issues/150440

--- a/qa/packaging/src/test/java/org/elasticsearch/packaging/test/DockerTests.java
+++ b/qa/packaging/src/test/java/org/elasticsearch/packaging/test/DockerTests.java
@@ -157,8 +157,10 @@ public class DockerTests extends PackagingTestCase {
      */
     public void test011SecurityEnabledStatus() throws Exception {
         waitForElasticsearch(installation, "elastic", PASSWORD);
-        final int statusCode = makeRequestAsElastic("wrong_password");
-        assertThat(statusCode, equalTo(401));
+        // Authenticating the reserved `elastic` user consults the security index. Right after startup that index can briefly be
+        // unavailable, in which case the request fails with 503 (UnavailableShardsException) rather than the expected 401. Retry
+        // until the reserved realm is serving authentication decisions. See https://github.com/elastic/elasticsearch/issues/150438
+        assertBusy(() -> assertThat(makeRequestAsElastic("wrong_password"), equalTo(401)));
     }
 
     /**

